runtime/service.go

```diff
@@ -354,6 +355,15 @@ func (s *service) StartShim(shimCtx context.Context, containerID, containerdBina
 		return "", err
 	}
 
+	// If there is a live socket, reuse the socket without calling CreateVM.
+	address, err := shim.SocketAddress(shimCtx, containerdAddress, s.vmID)
+	if err != nil {
+		return "", err
+	}
+	if shim.CanConnect(address) {
+		return address, nil
+	}
```

Comment:
I would double-check this part. While this code itself makes sense, I don't know why it *was* working when the sockets were abstract.

Reply:
This change is not needed. So I've removed from the last revision.